### DISCUSSION TITLE
DAW presets not recalled/working with latest AG plugin

### DISCUSSION THREAD

**Discussion by Henri75** [apohl79/audiogridder#992] [General]
So I noticed that on both cubase and studio one (mac) : 
When I try to recall a preset from within the DAW, nothing happens in the AG plugin.

But, If I add any plugin (whatever it be) with + in AG, then the plugin from the preset appears instead of the one I picked.
Same for a whole chain with X plugins : nothing load in AG plugin, but if I add any X random plugins, then the plugins from the DAW preset appears !

Looks like an initialization issue during preset recall from DAW

It seems it happened with all 1.2 beta up to the latest

Would be really usefull to fix that issue :)

**Comment by apohl79** [comment]
Issue created, thx for reporting this!

**Comment by apohl79** [comment]
As I can't reproduce this on my system, could you provide the plugin logs and the preset file that you are trying to load? Thx

**Comment by Henri75** [comment]
here is a simple example with a preset of one plugin blue cat gain (free).

When I reload the preset it is empty in AG !
but when i load any plugin, it restores the correct Nth entry for the nth
plugin you load.
In my exemple, the preset was for blue cat gain, after reloading the preset
AG is empty, then I load mv2 from waves and it loaded blue cat gain from
the preset ... but display mv2 in the list ! see attached screenshot.

Hope you can fix that one quickly ) It is really useful to have working
preset and recall !


Le mer. 24 août 2022 à 00:40, Serge Albou ***@***.***> a
écrit :

**Comment by Henri75** [comment]
I forgot to add the studio one preset (single plugin in AG : blue cat gain)

Le mer. 24 août 2022 à 01:33, Serge Albou ***@***.***> a
écrit :

**Reply by apohl79** [reply]
You will have to upload the attachments via browser.

**Comment by Henri75** [comment]
![Screen Shot 2022-08-24 at 01 26 44](https://user-images.githubusercontent.com/56123216/186732236-8f780f28-4eab-495c-8edb-90c47b68eed4.png)
[AudioGridderTray_2022-08-24_01-25-26.log](https://github.com/apohl79/audiogridder/files/9427067/AudioGridderTray_2022-08-24_01-25-26.log)
[GAIN.preset.zip](https://github.com/apohl79/audiogridder/files/9427071/GAIN.preset.zip)

**Comment by Henri75** [comment]
I reuploaded the screenshot, log and studio one preset

**Comment by apohl79** [comment]
Ok maybe I'm misunderstanding you. The GAIN preset file is no AG preset.

When you say that you recall a preset, you are not using the AG mechanism, but the one provided by the DAW?

**Comment by Henri75** [comment]
Correct, it is a a studio one preset !
Issue is when you save an AG preset inside the DAW presets, then recall the
preset from the DAW preset management (you load AG, then recall a preset
save in the DAW for that plugin)

Le jeu. 25 août 2022 à 19:45, Andreas Pohl ***@***.***> a
écrit :

**Reply by apohl79** [reply]
I can reproduce it now, thx for reporting this!

**Reply by apohl79** [reply]
The problem is that the plugin UI does not get updated when you load the daw present. Until the next version you have to close the AG plugin and open it again to see the correct remote plugins loaded.

**Comment by Henri75** [comment]
ok.
Will you fix it in the very next update ?

Le ven. 26 août 2022 à 13:30, Andreas Pohl ***@***.***> a
écrit :

**Reply by apohl79** [reply]
yes, the fix is already in and will be part of the next update

**Comment by Henri75** [comment]
Great ! thank you !

Le ven. 26 août 2022 à 17:04, Andreas Pohl ***@***.***> a
écrit :

**Comment by Henri75** [comment]
I just saw that you added in the source the ability to change/pass and ID
when launching the server right ?
Which means you can launch 2 servers with 2 differents ID ?
Will you add a per server ID directory for all AG files (config etc ...) ?
It will allow to run servers with different config (for example on mac 1
under Rosetta on 1 in native mode).



Le ven. 26 août 2022 à 17:04, Andreas Pohl ***@***.***> a
écrit :

**Reply by apohl79** [reply]
The possibility to run multiple servers by passing different IDs via commandline existed for a long time. But in the new version, you can autostart multiple servers with different IDs via config (and configure it in the server settings). Also since a few versions, the server has separate settings files per id. 

So yes you can run a rosetta server and an arm server on the same machine with different IDs and different settings. But you will not be able to autolaunch servers with different CPU architectures via config. You will still have to run (one of) the servers manually or use a script.

**Comment by Henri75** [comment]
Ok it is clear.

I still think you should allow per server full configuration and ressources
files (one server ID = one directory). For example using rosetta and M1
servers the plugins and blacklists cant be the same (intel plugins  running
under rosetta wont run in M1, and M1 ones wont run on rosetta) ! And having
separate logs per server is useful too.



Le ven. 26 août 2022 à 21:12, Andreas Pohl ***@***.***> a
écrit :

**Reply by apohl79** [reply]
thats already the case, but i don't want separate directories, so each server id has its own set of files..

**Comment by Henri75** [comment]
So I tried it and it doesnt seem to work.

I launched AG with commande line to pass -id  1 in rosetta mode (as I am on
M1)

then it launches in a couple of minutes, scanning a few plugins. This
shouldnt be the case if this instance use separate config files for plugins
etc ... Scanning should be a couple of hours !
So it seems to just use the same config files than the base instance
(without -id) instead of rescanning the plugins for this new -1 instance.

on M1 Mac, it is impossible to reuse the plugin cache as the plugins
wont be the same (M1 ones wont run in rosetta, intel wont run in native
mode).

May be I get something wrong ?

Thanks for your help

Le sam. 27 août 2022 à 13:15, Andreas Pohl ***@***.***> a
écrit :

**Comment by apohl79** [comment]
When you start a server with a new ID the very first time, it inherits the default config, plugin cache etc (of the server with ID 0) initially. After that it would only read its own files on the next restart. So after running your new server with ID 1, open the settings, change what is needed and do a wipe&rescan. 

Hope that helps.
